yi-misc-modes/src/Yi/Modes.hs

```diff
@@ -138,6 +139,11 @@ gnuMakeMode = styleMode GNUMake.lexer
               matches "GNUmakefile" = True
               matches filename      = extensionMatches [ "mk" ] filename
 
+markdownMode :: TokenBasedMode StyleName
+markdownMode = styleMode Markdown.lexer
+  & modeNameA .~ "markdown"
+  & modeAppliesA .~ anyExtension [ "md", "markdown", "mdown", "mkdn", "mdwn", "mkd" ]
```

Comment:
Add file extensions by referring to vim's markdown.
https://github.com/plasticboy/vim-markdown/blob/a3169545f330ec8080244c3ad755bb2211eca8a0/ftdetect/markdown.vim#L2